Assets/Scripts/NPC/Grid.cs

```diff
@@ -22,4 +43,79 @@ public Waypoint GetRandomWaypoint()
         var randomIndex = Random.Range(0, waypoints.Count);
         return waypoints[randomIndex];
     }
+
+    private void CalculateWaypoints()
+    {
+        waypointAmount.x = (int)Mathf.Abs(gridSize.x);
+        waypointAmount.y = (int)Mathf.Abs(gridSize.y);
+    }
+
+    private void InitPosition()
+    {
+        ResetXPosition();
+        ResetYPosition();
+    }
+
+    private void ResetXPosition()
+    {
+        initPosition.x = -(gridSize.x - 1) / 2;
+        safeZone.x = -initPosition.x - initPosition.x + 1;
+    }
+
+    private void ResetYPosition()
+    {
+        initPosition.y = (gridSize.y - 1) / 2;
+        safeZone.y = initPosition.y - -initPosition.y + 1;
+    }
```

Comment:
```suggestion
    [SerializeField] private float safezonePadding; // of safezoneMargin (heb ff geen idee wat precies de context is)

    private void ResetXPosition()
    {
        initPosition.x = -(gridSize.x - 1) / 2;
        safeZone.x = -initPosition.x - initPosition.x + safezonePadding;
    }

    private void ResetYPosition()
    {
        initPosition.y = (gridSize.y - 1) / 2;
        safeZone.y = initPosition.y - -initPosition.y + safezonePadding;
    }
```
Zo een instelbare safezone area hebben. (natuurlijk die nieuwe float bovenin het script)